R/commonAnovaBayesian.R

```diff
@@ -1749,9 +1788,11 @@ BANOVAcomputMatchedInclusion <- function(effectNames, effects.matrix, interactio
   on.exit(jaspBase:::assignFunctionInPackage(originalFun, "makeChainNeater", "BayesFactor"))
   jaspBase:::assignFunctionInPackage(.BANOVAmakeChainNeater, "makeChainNeater", "BayesFactor")
 
+  samplingIssues <- list()
   startProgressbar(nmodels, gettext("Sampling posteriors"))
   for (i in seq_len(nmodels)) {
-    if (is.na(reuseable[i])) {
+    # check if the state is reuseable and if it's not NULL, which means it didn't crash
+    if (is.na(reuseable[i]) || is.null(state$statistics[[reuseable[i]]])) {
```

Comment:
So @TimKDJ the issue you found was a side effect of better (but obviously not good enough) error handling when some models fail. Previously the assumption was that *all* models would be reuseable if there was a state, but this is no longer true since it's now possible that some models fail. In that case however, `state$statistics[[reuseable[i]]]` is `NULL` and this part (lines 1859-1860 of the new code) would throw an error.
```r
statistics[[i]] <- state$statistics[[reuseable[i]]] # assign NULL to list element
nms <- statistics[[i]]$names # crashes.
```
Now we check if 1. a model is reuseable (its formula is among those previously computed), and 2. the state of posterior samples for that model actually contains information. If a model failed in a previous run, we could copy the empty results for a particular model as it's probable that it will just fail again. Right now it just tries again, which I think is also okay albeit probably a bit slower. This is an extreme edge case though.